docs/src/explanations/events.md

```diff
@@ -397,3 +398,37 @@ record(scene, "test.mp4"; framerate = fps) do io
     end
 end
 ```
+
+## Tick Events
+
+Tick events are produced by the renderloop in GLMakie and WGLMakie, as well as `Makie.save` and `Makie.record` for all backends. They allow you to synchronize tasks such as animations with rendering. A Tick contains the following information:
+
+- `state::Makie.TickState`: Describes the situation in which the tick was produced. These include:
+    - `Makie.UnknownTickState`: A catch-all for uncategorized ticks. Currently only used for initialization of the tick event.
+    - `Makie.PausedRenderTick`: A tick originating from a paused renderloop in GLMakie. (This refers to the last attempt to draw a frame.)
+    - `Makie.SkippedRenderTick`: A tick originating from a `render_on_demand = true` renderloop in GLMakie where the last frame has been reused. (This happens when nothing about the displayed frame has changed. For animation you may consider this a regular render tick.)
+    - `Makie.RegularRenderTick`: A tick from a renderloop where the last frame has been redrawn.
+    - `Makie.OneTimeRenderTick`: A tick produced before generating an image for `Makie.save` and `Makie.record`.
+- `count::Int64`: Number of ticks produced since the first. During `record` this is relative to the first recorded frame.
+- `time::Float64`: The time that has passed since the first tick in seconds. During `record` this is relative to the first recorded frame and increments based on the `framerate` set in record.
+- `delta_time::Float64`: The time that has passed since the last tick in seconds. During `record` this is `1 / framerate`.
+
+For an animation you will generally not need to worry about tick state. You can simply update the relevant data as needed.
+```julia
+on(events(fig).tick) do tick
+    # For a simulation you may want to use delta times for updates:
+    position[] = position[] + tick.delta_time * velocity
+
+    # For a solved system you may want to use the total time to compute the current state:
+    position[] = trajectory(tick.time)
+
+    # For a data record you may want to use count to move to the next data point(s)
+    position[] = position_record[mod1(tick.count, end)]
+end
+```
+
+For an interactive figure this will produce an animation synchronized with real time. Within `record` the tick times match up the set `framerate` such that the animation in the produced video matches up with real time. 
+
+The only exception here is WGLMakie which currently runs the render loop during recording. Because of that `RegularRenderTick` and `OneTimeRenderTick` will mix during `record`, with the former being based on real time and the latter based on video time. If you do not restrict to `OneTimeRenderTick` here the resulting video will run faster than real time with `tick.delta_time` and may jump with `tick.time` and `tick.count`.
```

Comment:
We could probably avoid this by eating up the regular render ticks during recording
```julia
cb = on(events(fig).tick, priority = typemax(Int)) do tick
    return Consume(tick.state != Makie.OneTimeRenderTick)
end
# record ...
off(cb)
```
but it's not clear to me how to do that safely. My first thought would be to have the creation of `VideoStream` start this and `save` end this, but that doesn't seem save to me. I.e. VideoStream being created doesn't necessarily mean only recording takes place and you could save another video after the first save...

Reply:
I added that to `record` and suggested doing this for the lower level functions in the docs for now.

Reply:
Ok I changed my mind and just attached the filtering to VideoStream and save/a finalizer. Mostly because otherwise the tick event order is a bit weird. 

In the previous version the tick updates triggered just before `colorbuffer()`. So the order of operations was:
```julia
# record(foo, ...)
foo()
tick_update()
colorbuffer()
```
You'd still get updates coming from `on(..., tick)` before the render, but you couldn't use `events(fig).tick` to get the current frames time or count in `foo()`. I change things around a bit so that `record` effectively becomes:
```julia
tick_update()
foo()
colorbuffer()
```
so you can. This makes the filtering more important though, because the update for tick N actually happens at the end of frame N-1 in the `recordframe!()` call. (The first tick happens on VideoStream creation)

I also adjusted ticks so that the first tick is at count and time = 0.